includes/woopay/class-woopay-session.php

```diff
@@ -1058,4 +1080,78 @@ private static function get_inner_block( $current_block, $inner_block_name ) {
 
 		return $current_block['innerBlocks'][ $inner_block_index ];
 	}
+
+	/**
+	 * Catches and logs errors that occur during WooPay checkout processing.
+	 * This is particularly important for third-party plugin compatibility issues
+	 * (e.g., calling methods that don't exist on the WooPay SessionHandler).
+	 *
+	 * This method sets up an error handler that will catch PHP errors and add
+	 * them as order notes for debugging purposes.
+	 *
+	 * @param \WC_Order $order The order being processed.
+	 * @return void
+	 */
+	public static function catch_woopay_checkout_errors( $order ) {
+		if ( ! self::is_request_from_woopay() || ! ( $order instanceof \WC_Order ) ) {
+			return;
+		}
+
+		// Store the order ID so the error handler can access it.
+		self::$checkout_error_order_id = $order->get_id();
+
+		if ( self::$is_error_handler_registered ) {
+			return;
+		}
+
+		// Register shutdown function to catch fatal errors and restore previous handler.
+		register_shutdown_function(
+			function () {
+				$error = error_get_last();
+				if ( ! $error || ! self::is_request_from_woopay() ) {
+					return;
+				}
+
+				// Only handle fatal errors.
+				if ( ! in_array( $error['type'], [ E_ERROR, E_PARSE, E_CORE_ERROR, E_COMPILE_ERROR, E_USER_ERROR ], true ) ) {
+					return;
+				}
+
+				// Log the fatal error.
+				if ( ! self::$is_handling_error ) {
```

Comment:
When would the error handling code run twice in the same request? I did notice it ran on two separate requests though.

Reply:
Makes sense as fatal errors only run once, fixed on https://github.com/Automattic/woocommerce-payments/pull/11216/commits/da9593d91280c74041f33b16e0ff1a4e384e9e1a.